skbio/sequence/_grammared_sequence.py

```diff
@@ -22,11 +22,79 @@
 from ._sequence import Sequence
 
 
-class IUPACSequence(with_metaclass(ABCMeta, Sequence)):
-    """Store biological sequence data conforming to the IUPAC character set.
+class GrammaredSequenceException(TypeError):
+    pass
+
+
+class GrammaredSequenceMeta(ABCMeta, type):
+    def __new__(mcs, name, bases, dct):
+        cls = super(GrammaredSequenceMeta, mcs).__new__(mcs, name, bases, dct)
+
+        # Only perform validation on classes that aren't abstract.
+        if (type(cls.default_gap_char) is not abstractproperty and
```

Comment:
Yeah I meant "abstract", thanks. Sorry, "interleaved" was a bad way to describe that. What I meant was interleaved in time. So if user of skbio was working on writing a subclass of `GrammaredSequence` in a TDD fashion, the first error they would get would be from ABC. Then they would add the required property, try again, but this time the error might be from one of the static checks, then once they fix that the next error might be ABC again. I think this might be confusing, whereas if we force everything to not be abstract before any of the static checks run, all the related errors will come together.

I can implement the one check. Just a thought, what do you think about looping through all the attributes of the class and making sure nothing is abstract, rather than trying to maintain a list?